src/query/functions/utils/heap.go

```diff
@@ -137,14 +149,27 @@ func (fh *FloatHeap) Reset() {
 }
 
 // Flush flushes the float heap and resets it. Does not guarantee order.
-func (fh FloatHeap) Flush() []ValueIndexPair {
+func (fh *FloatHeap) Flush() []ValueIndexPair {
```

Comment:
Could you check if we still need this unordered `Flush`? I see it used in one place, but I don't know if it OK for unordered to be used there.

Reply:
We could safely remove unordered flush but for non instant queries, order doesn't matter and I feel that unordered flush  could be a more efficient to use in such cases.

Reply:
OK, makes sense then.